assets/app/scripts/services/ws.js

```diff
@@ -75,4 +75,16 @@ angular.module('openshiftConsole')
 
     return $ws;
   };
+})
+
+/* A WebSocket factory for kubernetesContainerTerminal */
+.factory("ContainerWebSocket", function(API_CFG, $ws) {
+  return function AuthWebSocket(url, protocols) {
+    var scheme;
+    if (url.indexOf("/") === 0) {
+      scheme = window.location.protocol === "http:" ? "ws://" : "wss://";
+      url = scheme + API_CFG.openshift.hostPort + url;
+    }
+    return $ws({ url: url, method: "WATCH", protocols: protocols });
```

Comment:
to automatically pick up auth token, you need to set an `auth` stanza in the options, like `auth:{}` (data.js adds this for any `$http`/`$ws` requests it generates